### ISSUE TITLE
Bitpacking default pack and unpack dimensions do not round-trip multidimensional tensors

### ISSUE THREAD

**Issue by aswanth-07** [pytorch/ao#4815]
### Bug

The optimized `pack()` and `unpack()` helpers have incompatible default dimensions: `pack(..., dim=-1)` packs the last dimension, while `unpack(..., dim=0)` expands the first dimension. As a result, the documented default round trip only works for one-dimensional tensors.

### Reproduction

```python
import torch
from torchao.prototype.dtypes.uintx.bitpacking import pack, unpack

x = torch.randint(0, 16, (8, 16), dtype=torch.uint8)
packed = pack(x, 4)
restored = unpack(packed, 4)

print(x.shape, restored.shape)
```

On current `main` (`36027d82d`), this prints:

```text
torch.Size([8, 16]) torch.Size([16, 8])
```

The CPU reference pair (`pack_cpu` / `unpack_cpu`) already uses `dim=-1` consistently, and TorchAO's internal `UIntxTensor` callers pass an explicit dimension.

### Proposed fix

Change `unpack()`'s default to `dim=-1` so the optimized pair has the same default contract as the CPU pair, and add a multidimensional round-trip regression covering bit widths 1 through 7. I have this small fix ready.
